model_signing/signing/manifest.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pathlib import Path
+from typing import Self
+from typing_extensions import override
+
+import dataclasses
+import json
+
+from model_signing.hashing import hashing
+from model_signing.manifest.manifest import ResourceDescriptor
+from model_signing.manifest.manifest import Manifest
+from model_signing.signature import signature
+from model_signing.signing import signing
+
+
+class ManifestPayload(signing.SigningPayload):
+
+    def __init__(
+            self, resource_descriptors: list[ResourceDescriptor]):
+        self._items: dict[str, hashing.Digest] = {
+            r.identifier: r.digest for r in resource_descriptors
+        }
+
+    @classmethod
+    @override
+    def from_manifest(cls, manifest: Manifest) -> Self:
+        descriptors = []
+        for d in manifest.resource_descriptors:
+            descriptors.append(d)
+        return cls(descriptors)
```

Comment:
I'm unsure about this. We're just building a dictionary out of items fed from another dictionary. And we also lose the ordering guarantee.

Why not make the payload directly be in-toto here?